opentelemetry-api/tests/test_backend.py

```diff
@@ -0,0 +1,67 @@
+from importlib import reload
+import sys
+
+import pytest
+
+from opentelemetry import backend
+from opentelemetry.trace import Tracer
+
+class DummyTracer(Tracer):
+    pass
+
+DUMMY_TRACER = None
+
+def get_opentelemetry_backend_impl(type_):
+    global DUMMY_TRACER #pylint:disable=global-statement
+    assert type_ is Tracer
+    DUMMY_TRACER = DummyTracer()
+    return DUMMY_TRACER
+
+#pylint:disable=redefined-outer-name,protected-access,unidiomatic-typecheck
+
+def test_get_default(backend=backend):
+    backend = reload(backend)
+    backend._UNIT_TEST_IGNORE_ENV = True
+    tracer = backend.tracer()
+    assert type(tracer) is Tracer
+
+def test_get_set(backend=backend):
+    backend = reload(backend)
+    set_tracer = DummyTracer()
+    backend.set_tracer(set_tracer)
+    tracer = backend.tracer()
+    assert tracer is set_tracer
+
+    # Set again
+    set_tracer = DummyTracer()
+    backend.set_tracer(set_tracer)
+    tracer = backend.tracer()
+    assert tracer is set_tracer
+
+def test_get_set_import_from(backend=backend):
+    backend = reload(backend)
+    get_tracer = backend.tracer # Simulate `import tracer from backend as get_tracer`
+    set_tracer = DummyTracer()
+    backend.set_tracer(set_tracer)
+    tracer = get_tracer()
+    assert tracer is set_tracer
+
+    # Set again
+    set_tracer = DummyTracer()
+    backend.set_tracer(set_tracer)
+    tracer = get_tracer()
+    assert tracer is set_tracer
+
+@pytest.mark.parametrize('envvar_suffix', ['TRACER', 'DEFAULT'])
+def test_get_envvar(envvar_suffix, monkeypatch, backend=backend):
```

Comment:
Pytest's fixture magic is one reason I prefer standard unittests, but I may be in the minority here.

Reply:
Yeah, pytest does indeed have lots of magic. Did you know it uses a custom module loader to load the test modules and instrument the asserts by rewriting the AST?

Reply:
Ha! I did not.

Reply:
I removed the unit tests for now -- resolving this.